### DISCUSSION TITLE
cart drawer not updating on the first add to cart

### DISCUSSION THREAD

**Discussion by Webiink** [EvgeniyMukhamedjanov/liquid-ajax-cart#71] [Q&A]
cart drawer not updating on the first add to cart I have to refresh it to see the item

**Comment by EvgeniyMukhamedjanov** [comment]
Share a preview link

**Reply by jnabong-kyna** [reply]
Hello @EvgeniyMukhamedjanov! I'm also experiencing this issue. When I quickly click the 'Add to Cart' button, the product isn't added to the cart drawer—it only opens the drawer. I have to refresh the page or click the 'Add to Cart' button again for the cart drawer to update properly. How can I fix this?

**Reply by EvgeniyMukhamedjanov** [reply]
Hey @jnabong-kyna ,
Thank you for reaching out!
Do you experience the same issue on the demo store https://liquid-ajax-cart.myshopify.com/
Also I would like to clarify what does "quickly click" means? Clicking multiple times or click as soon as the button appear on the page?

**Reply by jnabong-kyna** [reply]
@EvgeniyMukhamedjanov I tried checking that demo store, and it seems to be working fine. And for 'quickly click,' it means clicking as soon as the button appears on the page.

**Reply by EvgeniyMukhamedjanov** [reply]
@jnabong-kyna Could you share the website, I'll check why it is happening in your case.

**Reply by jnabong-kyna** [reply]
@EvgeniyMukhamedjanov here's the website https://komfortkissen.de/

**Reply by jnabong-kyna** [reply]
Hello @EvgeniyMukhamedjanov have you checked the website? Any update, please?

**Reply by EvgeniyMukhamedjanov** [reply]
Hello @jnabong-kyna ,
sorry for the delayed answer.
I checked the website this is what i see:

https://github.com/user-attachments/assets/32a3cd5f-7c71-4405-8657-ca94e842d161

I set up site speed to 3G to simulate slow connection. Once the button appears on the page, I clicked it and inspected the request it initiated. After some delay, the cart is opened and the product was there. So I couldn't replicate the issue. The delay between the click and "product is added to cart" is due to a lot of other scripts/images/requests working on page load, this is not something related to LAC. Also I noticed, you have the <product-form> component wrapping the <ajax-cart-product-form> component. I guess the <product-form> is coming from the Dawn theme and also responsible for adding a product to the cart? In this case I would suggest getting rid of it as it seems like there are two scripts are trying to add a product to the cart.
